### ISSUE TITLE
UI & security config - spring failures

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#1812]
I am seeing errors in various situations from the UI around:
```
Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2019-10-22 10:53:42.561 ERROR 49526 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'subjectAreaProjectController': Unsatisfied dependency expressed through field 'authService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'securityConfig': Unsatisfied dependency expressed through field 'authService'; nested exception is org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'token': Requested bean is currently in creation: Is there an unresolvable circular reference?
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:596) ~[spring-beans-5.1.8.RELEASE.jar!/:5.1.8.RELEASE]
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:90) ~[spring-beans-5.1.8.RELEASE.jar!/:5.1.8.RELEASE]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:374) ~[spring-beans-5.1.8.RELEASE.jar!/:5.1.8.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1411) ~[spring-beans-5.1.8.RELEASE.jar!/:5.1.8.RELEASE]
```

It seems to occur after some builds and not others - so I need to investigate further.

I initially saw after updating the h2 database version - which I then backed off, but it's occurred here again - this time whilst looking at updating some apache commons versions.

What is odd is that the exact same error has occurred in a variety of different configurations which may suggest some underlying concern.

Opening in case anyone else sees something different. I will investigate a little further, but I note there have been very recent changes in the UI & security configuration, perhaps there is some fragility there.

I am also inclined to test/.apply the latest spring updates & then revisit

**Comment by ruxandraRosu** [comment]
Hi Nigel,
Is there something different in the build configuration or do you mean it is randomly happening?

**Comment by planetf1** [comment]
Not random, but related to build configuration each time as I've been working through testing various dependency updates. But it doesn't seem entirely consistent. 

I am going to close until I see a reproducible pattern as currently it's not actionable

**Comment by planetf1** [comment]
I'm continuing to see this, yet have not been able to identify a specific pattern. This is a concern.

I will do some more testing from a containerised build (uses assets from official build) rather than build locally. 

Re-opening until understood - the general appearance is that of fragility in this area - not one specific change seems to cause it.

cc: @davidradl

**Comment by ruxandraRosu** [comment]
Hi Nigel,
This bug is from my change, I am looking at it.
Thanks

**Comment by planetf1** [comment]
I believe this was fixed by #1794 . closing (@ruxandraRosu let me know if this is wrong)
